fixtures/load-repo.js

```diff
@@ -0,0 +1,205 @@
+  const files = await walk(name, { filesonly: true, cwd: reposDir });
+
+  console.log(kleur.bold().blue('Looking for files to lint'));
+
+  const interestingFiles = await Promise.all(
+    files
+      .filter((f) => f.endsWith('.js') && !f.replace(name, '').includes('test'))
+      .map(async (file) => {
+        const contents = await readFileAsync(path.join(reposDir, file), 'utf8');
+        return { file, contents };
+      })
+  );
+
+  const eslint = new CLIEngine({
+    configFile: path.join(fixturesDir, '.eslintrc.js'),
+    fix: true,
+  });
+
+  const filesWithNumNodeTypes = await Promise.all(
+    interestingFiles.map(async ({ file, contents }) => {
+      const seenNodeTypes = new Set();
+      let numNodes = 0;
+      let parsed;
+      /**
+       * This should be returned if the file is invalid
+       * This could happen if there is a parsing error or if it is empty
+       * or if it has too many linting errors, etc
+       */
+      const invalidResult = {
+        file,
+        contents,
+        score: 0,
+        numLintErrors: 0,
+        numNodeTypes: 0,
+        loc: 0,
+      };
+      const filenameRelative = path.join(reposDir, file);
+      try {
+        // Attempts to parse each file with babel
+        // Traverses the file and counts up the number of node types and number of nodes
+        // Files with more node types tend to be more interesting and better candidates for lint testing
+        parsed = await babel.parseAsync(contents, {
+          parserOpts: {
+            plugins: [
+              'jsx',
+              'classProperties',
+              'dynamicImport',
+              'nullishCoalescingOperator',
+              'optionalChaining',
+            ],
+          },
+          filenameRelative,
+        });
+      } catch (error) {
+        return invalidResult;
+      }
+
+      babel.traverse(parsed, {
+        enter(path) {
+          numNodes++;
+          seenNodeTypes.add(path.type);
+        },
+      });
+
+      const numNodeTypes = seenNodeTypes.size;
+
+      let eslintReport;
+      try {
+        eslintReport = eslint.executeOnText(contents).results[0];
+      } catch (error) {
+        console.log('eslint error', file, error);
+        return invalidResult;
+      }
+
+      const numLintErrors = eslintReport.errorCount + eslintReport.warningCount;
+      const eslintFixedContents = eslintReport.output || contents;
+
+      let prettifiedContents;
+
+      try {
+        prettifiedContents = prettier.format(eslintFixedContents, prettierOpts);
+      } catch (error) {
+        console.log('prettier error', file, error);
+        return invalidResult;
+      }
+
+      // Generates a "score" to a file based on the # of nodes, # of node types
+      // Used for sorting
+      // NumNodeTypes is weighted much more strongly
+      // because files tend to have a lot of nodes compared to the number of node types
+      const score = numNodeTypes * 10 + numNodes / 1000 - numLintErrors * 30;
```

Comment:
Should any of this (the metrics values) be configurable?

Reply:
I think editing these values is a reasonable way to configure it

Reply:
@calebeby, OK, can you at least hint how you came up with these numbers? They seem magic to me, and it would be good to know why they are what they are and why anyone might want to tweak them.

Reply:
Also, I'm having a little trouble understanding the formula here. It's not assessing the number of nodes in relation to the number of node types, but just adding them together and then subtracting errors (each with its own weight multiplier, which is the part I get). This confuses me. I'm sure you understand why you're doing it this way, but I could use a little more explanation.

Reply:
> but just adding them together and then subtracting errors (each with its own weight multiplier, which is the part I get). This confuses me

Having more nodes increases the ranking. Having more node types increases the ranking as well. Having more lint errors decreases the ranking. numLintErrors is subtracted, so that the score goes down when the lint errors increase. Not sure whether that answers the part you are confused about.

The numbers are made up. They are weighting factors. I chose a number, tried it out, and changed it until I was happy with the way files were being ranked. 

Here was my process:
Initially, the weighting on numNodes and numNodeTypes was equal. I noticed that the number of nodes and the number of lint errors had nearly no effect on the ranking, they were pretty much only being sorted by the size of the file/number of nodes. This was happening because the files that were being sorted often had many thousands of nodes but only thirty-something distinct node types. So I adjusted the weightings until it seemed like the sorting was about right.

Reply:
Thanks, @calebeby, that does help clarify how you came to this formula. It sounds to me like you didn't reference any other code to come up with this idea, but if you did, adding (a) reference(s) would be nice, but otherwise, no further questions.

Reply:
👍 No references 😄 

Was there any part of my explanation that you think should be in code comments?

Reply:
The formula choices could be documented with a bit more of this detail, in case anyone else is ever in the place of needing to work with this file. Just some clarification that this entire approach may be a Caleb original, and therefore how you arrived at the choices you made could be helpful (as it has been here). Thanks